models/omop/measurement.sql

```diff
@@ -57,7 +57,7 @@ WITH snomed_measurements AS (
         , 32827 AS measurement_type_concept_id
         , 0 AS operator_concept_id
         , CASE
-            WHEN o.observation_value ~ '^[-+]?[0-9]+\.?[0-9]*$'
+            WHEN {{ regexp_like("o.observation_value", "^[-+]?[0-9]+\.?[0-9]*$") }}
                 THEN {{ dbt.cast("o.observation_value", api.Column.translate_type("decimal")) }}
             ELSE {{ dbt.cast("null", api.Column.translate_type("decimal")) }}
         END AS value_as_number
```

Comment:
the new macro is useful and we should keep it for future use (thanks!!!) - but looking at this gross regex, maybe we could use `safe_cast` here instead?  https://docs.getdbt.com/reference/dbt-jinja-functions/cross-database-macros#safe_cast

Reply:
@katy-sadowski Had another look through the cross-DB macros and fairly sure there's no regular expression oriented ones sadly - I don't think there is a way around this without using custom dispatchers / macros like this

It raises the question of _do_ we event want to add ❄️ support! I think it is nice to have - just wish they used the `~` operator...

I looked into transpiring with SQLGlot, but I think thats a lot of work for relatively low yield - pulling out the DAG from `dbt` etc as per here https://github.com/OHDSI/dbt-synthea/issues/39#issuecomment-2393032632 is quite a lot of work which isn't in the scope of the project (IMHO!)

Reply:
ahhh sorry i forgot about what you already said in https://github.com/OHDSI/dbt-synthea/issues/68 about safe_cast not being supported on all platforms.  let's carry on with the regex approach!  agree re: sqlglot.